Remotion/Data/DomainObjects/Persistence/Rdbms/SqlServer/Parameters/SqlTableValuedParameterValue.cs

```diff
@@ -82,5 +82,84 @@ IEnumerator<SqlDataRecord> IEnumerable<SqlDataRecord>.GetEnumerator ()
   }
 
   public int Count => _records.Count;
+
+  public bool Equals (SqlTableValuedParameterValue? other)
+  {
+    if (other == null)
+      return false;
+
+    if (TableTypeName != other.TableTypeName)
+      return false;
+
+    if (_columnMetaData.Length != other._columnMetaData.Length)
+      return false;
+
+    if (_records.Count != other._records.Count)
+      return false;
+
+    for (var i = 0; i < _columnMetaData.Length; i++)
+    {
+      var ours = _columnMetaData[i];
+      var theirs = other._columnMetaData[i];
+      if (!theirs.Name.Equals(ours.Name))
+        return false;
+
+      if (!theirs.DbType.Equals(ours.DbType))
+        return false;
+
+      if (!theirs.MaxLength.Equals(ours.MaxLength))
+        return false;
+    }
+
+    for (int rec = 0; rec < _records.Count; rec++)
+    {
+      var ours = _records[rec];
+      var theirs = other._records[rec];
+      if (ours.FieldCount != theirs.FieldCount)
+        return false;
+
+      for (int col = 0; col < ours.FieldCount; col++)
+      {
+        var ourValue = ours[col];
+        var theirValue = theirs[col];
+        if (!ourValue.Equals(theirValue))
```

Comment:
```suggestion
        if (!ourValue.Equals(theirValue))
```
```suggestion
        if (!ourValue == theirValue)
```

Reply:
These are `object`-typed variables, so I think using `.Equals()` is safer at this point.